cli/types/cypress.d.ts

```diff
@@ -518,7 +518,7 @@ declare namespace Cypress {
     off: Actions
   }
 
-  type CanReturnChainable = void | Chainable
+  type CanReturnChainable = void | Chainable | Promise<unknown>
```

Comment:
I know this PR is already merged, but I had a question on this. There are some commands that are failing for me as well on Cypress v5.1.0. And I came across this PR as the fix for a similar issue that I was facing. The problem resolves for me if I do the following in my local `cypress.d.ts` file:
```typescript
  type CanReturnChainable = void | Chainable | boolean | string | number | string[] | undefined
```
Please let me know if this seems to be the correct solution and if I should open a PR for this? :)
Thank you in advance for your response!
cc: @sainthkh @Bkucera 

The code snippet that's causing this type error:
```typescript
/** Get the value of a DataTable date filter */
Cypress.Commands.add(
  'getDateValue',
  { prevSubject: 'element' },
  ($filter: JQuery<HTMLElement>) => {
    const input = $filter.find(DATE_FILTER_INPUT_SELECTOR);
    return input.val();
  },
);
```

Reply:
Is it used like below?

```js
const x = cy.getDateValue();
```

Reply:
Something similar, it is used as
```typescript
cy.getDateFilter('Min Due Date')
  .toggleEnabled()
  .setDateValue(minDueDate);
```

Reply:
Conventionally, Cypress commands return `thenable` objects. But in your case, `boolean | string | number | string[] | undefined` are not `thenable`. Because of that, I cannot tell you for sure that they are OK.

@BKucera. Is it OK to return non-`thenable`s?

Reply:
I see! Thank you! I appreciate your response. I'll wait for @Bkucera 's response on this thread to see what he says.